src/Components/SignIn.js

```diff
@@ -13,20 +14,63 @@ const validation = Yup.object().shape({
     .required('Must enter a password')
 })
 
+const SIGN_IN = gql`
+  mutation sign_in($data: SigninInput!){
+    signin(data: $data){
+      user{
+        id
+      }
+      jwt
+    }
+  }
+`
+
 export default function SignIn () {
+  const [error, setError] = useState(null)
+
+  const [signin] = useMutation(SIGN_IN, {
+    onCompleted: LoginSuccess,
+    onError: LoginFailed
+  }
+  )
+
+  function LoginSuccess () {
+    return (
+      setError(false)
+    )
+  }
+
+  function LoginFailed () {
+    return (
+      setError(true)
+    )
+  }
+
   return (
     <div className='outer'>
       <div className='middle'>
         <div className='inner'>
           <div className='wrapper fadeInDown'>
             <div id='formContent'>
               <h1>Sign In</h1>
+              {error &&
+                <h3 className='error'>  oh snap! Something went wrong  </h3>}
+              {error === false &&
+                <h3 className='correct'>  Log in successfully </h3>}
               <Formik
                 initialValues={{ username: '', password: '' }}
                 validationSchema={validation}
-                onSubmit={(values, { setSubmitting, resetForm }) => {
+                
+                onSubmit={async(values, { setSubmitting, resetForm }) => {
                   setSubmitting(true)
-                  alert(JSON.stringify({ values }))
+                  const response = await signin({
+                    variables: {
+                      data: values
+                    }
+                  })
+                  localStorage.setItem('token', response.data.signin.jwt)
+                  localStorage.setItem('user', response.data.signin.user.id)
```

Comment:
- This is not the best place for this. Extract the functionality of storing and reading the token to its own file, so that you can reuse it both on the login and the signup form.
- No need to store the user id on the local storage. You can either extract it from the token or use the `currentUser` query to obtain the `id`.